test/e2e/idle_connection_test.go

```diff
@@ -236,12 +245,35 @@ func idleConnectionCreatePod(ctx context.Context, namespace, name, image string,
 			},
 		},
 	}
+}
 
-	if err := kclient.Create(ctx, pod); err != nil {
-		return nil, fmt.Errorf("failed to create pod %s/%s: %w", pod.Namespace, pod.Name, err)
+func idleConnectionCreateReplicaSet(ctx context.Context, namespace, name, image string, labels map[string]string) (*appsv1.ReplicaSet, error) {
+	pod := idleConnectionBuildPod(namespace, name, image, labels)
+	one := int32(1)
+	rs := &appsv1.ReplicaSet{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      name,
+			Namespace: namespace,
+		},
+		Spec: appsv1.ReplicaSetSpec{
+			Replicas: &one,
+			Selector: &metav1.LabelSelector{
+				MatchLabels: pod.Labels,
+			},
+			Template: corev1.PodTemplateSpec{
+				ObjectMeta: metav1.ObjectMeta{
+					Labels: pod.Labels,
+				},
+				Spec: pod.Spec,
+			},
+		},
 	}
 
-	return pod, nil
+	if err := kclient.Create(ctx, rs); err != nil {
```

Comment:
do you wanna retry here for the sake of network problems? (as we've been doing on other PRs)